test/xpu_api/numerics/complex.number/cases.h

```diff
@@ -218,48 +215,51 @@ CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
         return non_zero_nan;
     }
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
 inline
 int
 classify(double x)
 {
-// Suppress clang warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
-
     if (x == 0)
         return zero;
     if (std::isinf(x))
         return inf;
     if (std::isnan(x))
         return NaN;
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
-void is_about(float x, float y)
+auto
+get_eps(float)
 {
-// Suppress clang warning: floating-point comparison is always true; constant cannot be represented exactly in type 'float' [-Wliteral-range]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_LITERAL_RANGE
-
-    assert(std::abs((x-y)/(x+y)) < 1.e-6);
+    constexpr float eps = 1.e-6;
+    return eps;
+};
 
-CLANG_DIAGNOSTIC_POP
-}
+template <typename T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, T>::type
+get_eps(T)
+{
+    static_assert(::std::is_same_v<T, double> || ::std::is_same_v<T, long double>);
+    constexpr T eps = 1.e-11;
+    return eps;
+};
 
-void is_about(double x, double y)
+template <typename T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, void>::type
+is_about(T x, T y)
 {
-    assert(std::abs((x-y)/(x+y)) < 1.e-14);
+    const auto exp = get_eps(T{});
+    assert(std::abs(x - y) < exp);
 }
 
-void is_about(long double x, long double y)
+template <typename T1, typename T2>
+void
+is_about_complex(const dpl::complex<T1>& x, const dpl::complex<T2>& y)
```

Comment:
I think such complex number comparison  is not quite correct...
See
https://stackoverflow.com/questions/34231008/how-to-compare-two-complex-numbers
https://math.stackexchange.com/questions/310931/comparing-complex-numbers

Reply:
In principal I agree with you.

But in these cases we initially had for example something like this:
```assert(dpl::pow(x, y) == dpl::pow(dpl::complex<V>(x, 0), dpl::complex<V>(y)));```
which were transformed by me into 
```is_about_complex(dpl::pow(x, y), dpl::pow(dpl::complex<V>(x, 0), dpl::complex<V>(y)));```

Because we change operation ```==``` to something other, I think we are able to use current implementation of ```is_about_complex``` function.

Reply:
Fixed